### ISSUE TITLE
clip method does not always return empty dataframe if intersection is empty

### ISSUE THREAD

**Issue by rwijtvliet** [geopandas/geopandas#1363]
The return object from the `geopandas.clip` method has inconsistent behaviour in the case, that all rows do not lie within the mask:

```python
mask = sg.box(0, 0, 10, 10)

#No issue: both points directly above the mask
gdf = gpd.GeoDataFrame(geometry=[sg.Point([5, 15]), sg.Point([5, 20])])
gpd.clip(gdf, mask) #returns empty dataframe, as expected
Out[10]: 
Empty GeoDataFrame
Columns: [geometry]
Index: []

#Also no issue: both points directly to the left; both points directly below; both points directly to the right

#Issue: one point directly above and the other directly to the right of the mask
gdf = gpd.GeoDataFrame(geometry=[sg.Point([5, 15]), sg.Point([20, 5])])
gpd.clip(gdf, mask) #should also return empty dataframe, but does not:
Out[20]: 
  geometry
0     None
1     None

#Also issue: one above&to right and other below&to left
```

I hope that helps - wasn't easy to pin down :)

Let me know if there is any other information I can contribute

**Comment by rwijtvliet** [comment]
If it's helpful: the error message I received when using my actual data (with the `gdf` being a lot longer and consisting of `shapely LineStrings` and `MultiLineStrings`) was `ValueError: Must pass array with one dimension only.`

**Comment by martinfleis** [comment]
Thank you for the report. This difference here is caused by the check we do before the actual clip operation. If bounding boxes of df and mask do not overlap, we return empty df:

https://github.com/geopandas/geopandas/blob/078062d303e401aaa5e37e04c0e7c3ce188920fe/geopandas/tools/clip.py#L151-L159

If they overlap, the result comes from the clip algorithm itself. You are right, that both results should be the same. This likely comes from sorting in the end, so we just need to add a check if `concat` is empty and skip sorting in that case here:

https://github.com/geopandas/geopandas/blob/078062d303e401aaa5e37e04c0e7c3ce188920fe/geopandas/tools/clip.py#L249-L256

It should be a simple fix. Would you like to make a PR fixing it?

**Comment by rwijtvliet** [comment]
I've never done such a thing but I'll give it a try. Many thanks for the quick response!

**Comment by martinfleis** [comment]
Closed by #1363
